### PR TITLE
[REF-2940] Added error handling for permissions, keys, and paths

### PR DESCRIPTION
**Pull request by Chucheen**

### CODE REVIEW

**Comment by Chucheen** [comment]
https://app.clubhouse.io/imagineer/story/2940/validate-api-key-and-check-for-proper-permissions-with-solovis-api

File: spec/solovis_api_client/site_spec.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require 'spec_helper'
 
 RSpec.describe SolovisApiClient::Site do
-  let(:client) { SolovisApiClient::Base.new(url: ENV['SOLOVIS_URL'], api_key: ENV['SOLOVIS_API_KEY']) }
+  let(:client) { SolovisApiClient::Base.new(url: 'https://synap-demo.gateway.solovis.com', api_key: '123__123') }
```
Comment (rfleury2) [inline spec/solovis_api_client/site_spec.rb]: Can you use the real credentials for `api_key`?  This fake credential stuff has been a bit of a headache, so I'd rather avoid it

Comment (Chucheen) [reply spec/solovis_api_client/site_spec.rb]: lol, changed man. Thanks for remembering!

**Comment by Chucheen** [comment]
@rfleury2 it's back to you :)

**Comment by rfleury2** [comment]
@Chucheen I tagged `0.1.10`.

**Pull request merged**